api/csharp/src/Microsoft.Azure.Devices.Proxy/Socket/SocketAddress.cs

```diff
@@ -605,4 +763,87 @@ public class SocketAddressCollection : SocketAddress, IEquatable<SocketAddressCo
                 Inner.GetHashCode();
         }
     }
+
+    /// <summary>
+    /// A socket address bound to a proxy
+    /// </summary>
+    [DataContract]
+    public class BoundSocketAddress : SocketAddress, IEquatable<BoundSocketAddress> {
+
+        [DataMember(Name = "family", Order = 1)]
+        public override AddressFamily Family => AddressFamily.Bound;
+
+        /// <summary>
+        /// Bound addresses
+        /// </summary>
+        [DataMember(Name = "local", Order = 2)]
+        public SocketAddress LocalAddress { get; set; }
+
+        /// <summary>
+        /// Host name to use
+        /// </summary>
+        [DataMember(Name = "remote", Order = 3)]
+        public SocketAddress RemoteAddress { get; set; }
+
+        /// <summary>
+        /// Default constructor
+        /// </summary>
+        public BoundSocketAddress() {
+        }
+
+        /// <summary>
+        /// Bind constructor
+        /// </summary>
+        public BoundSocketAddress(SocketAddress localAddress, SocketAddress remoteAddress) {
+            LocalAddress = localAddress;
+            RemoteAddress = remoteAddress;
+        }
+
+        /// <summary>
+        /// Comparison
+        /// </summary>
+        /// <param name="that"></param>
+        /// <returns></returns>
+        public bool Equals(BoundSocketAddress that) {
+            if (that == null) {
+                return false;
+            }
+            return
+                this.LocalAddress.Equals(that.LocalAddress) &&
+                this.RemoteAddress.Equals(that.RemoteAddress);
+        }
+
+        /// <summary>
+        /// Comparison
+        /// </summary>
+        /// <param name="that"></param>
+        /// <returns></returns>
+        public override bool Equals(object that) {
+            return Equals(that as BoundSocketAddress);
+        }
+
+        /// <summary>
+        /// Returns the remote address as a proxy address
+        /// </summary>
+        /// <returns></returns>
+        public override ProxySocketAddress AsProxySocketAddress() =>
+           RemoteAddress.AsProxySocketAddress();
+
+        /// <summary>
+        /// Stringify address
+        /// </summary>
+        /// <returns></returns>
+        public override string ToString() {
+            return RemoteAddress.ToString();
+        }
+
+        /// <summary>
+        /// Returns hash for efficient lookup in list
+        /// </summary>
+        /// <returns></returns>
+        public override int GetHashCode() {
+            return (RemoteAddress.GetHashCode() * 31) ^ 
```

Comment:
Pseudo hashing.  I need to replace this with something better eventually.  Will open a bug.